src/pudl/transform/ferc1.py

```diff
@@ -849,6 +849,33 @@ class PlantsPumpedStorageFerc1TableTransformer(Ferc1AbstractTableTransformer):
     table_id: Ferc1TableId = Ferc1TableId.PLANTS_PUMPED_STORAGE_FERC1
 
 
+class PurchasedPowerTableTransformer(Ferc1AbstractTableTransformer):
+    """Transformer class for ``purchased_power_ferc1`` table."""
+
+    table_id: Ferc1TableId = Ferc1TableId.PURCHASED_POWER_FERC1
+
+    @cache_df(key="main")
+    def transform_main(self, df: pd.DataFrame) -> pd.DataFrame:
+        """Table-specific transforms for the ``purchased_power_ferc1`` table.
+
+        Perform all default :meth:`Ferc1AbstractTableTransformer.transform_main`
+        procedures, plus drop duplicates and run the encoder.
+        """
+        df = (
+            super()
+            .transform_main(df)
+            # Drop records containing no useful data and also any completely duplicate
+            # records -- there are 6 in 1998 for utility 238 for some reason...
+            .drop_duplicates()
+            .pipe(
+                pudl.metadata.classes.Package.from_resource_ids()
+                .get_resource(self.table_id.value)
+                .encode
```

Comment:
Oooh, our first encoding on FERC data?

If you're using the encoder on this table shouldn't there also be a new encoding table defined with the acceptable values, and don't the foreign key relationships that cause the encoding need to be defined for the resource?

Reply:
this table was being encoded  in the deleted function... but i was wondering why it was being used at all.

Reply:
the `purchase_type_code` is the column being encoded